js/src/components/modals/RedirectUpsell.js

```diff
@@ -0,0 +1,178 @@
+		min-width: 0;
+		width: 86vw;
+	}
+`;
+
+const StyledIcon = styled( Icon )`
+	float: ${ getRtlStyle( "right", "left" ) };
+	margin: ${ getRtlStyle( "0 0 16px 16px", "0 16px 16px 0" ) };
+
+	&& {
+		width: 150px;
+		height: 150px;
+
+		@media screen and ( max-width: 680px ) {
+			width: 80px;
+			height: 80px;
+		}
+	}
+`;
+
+/**
+ * Returns the UpsellBox component.
+ *
+ * @returns {ReactElement} The UpsellBox component.
+ */
+class RedirectUpsell extends React.Component {
+	/**
+	 * The constructor.
+	 *
+	 * @param {object} props The component props.
+	 *
+	 * @returns {void}
+	 */
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			modalIsOpen: false,
+		};
+
+		this.openModal  = this.openModal.bind( this );
+		this.closeModal = this.closeModal.bind( this );
+
+		jQuery( ".wpseo-open-gsc-redirect-modal" ).click(
+			/**
+			 * Onclick event.
+			 *
+			 * @param {Event} event The event.
+			 *
+			 * @returns {void}
+			 */
+			( event ) => {
+				event.preventDefault();
+
+				this.openModal();
+			}
+		);
+	}
+
+	/**
+	 * Sets the modal is open state to true.
+	 *
+	 * @returns {void}
+	 */
+	openModal() {
+		this.setState( {
+			modalIsOpen: true,
+		} );
+	}
+
+	/**
+	 * Sets the modal is open state to false.
+	 *
+	 * @returns {void}
+	 */
+	closeModal() {
+		this.setState( {
+			modalIsOpen: false,
+		} );
+	}
+
+	/**
+	 * Renders the component.
+	 *
+	 * @returns {ReactElement} The rendered element.
+	 */
+	render() {
+		if ( this.state.modalIsOpen === false ) {
+			return null;
+		}
+
+		const heading = sprintf(
+			/* translators: %s expands to 'Yoast SEO Premium'. */
+			__( "Get %s", "wordpress-seo" ),
+			"Yoast SEO Premium"
+		);
+
+		const title = sprintf(
+			/* translators: %1$s expands to a 'strong' start tag, %2$s to a 'strong' end tag. */
+			__( "Creating redirects is a %s feature", "wordpress-seo" ),
+			"Yoast SEO Premium"
+		);
+
+		const intro = sprintf(
+			/* Translators: %1$s: expands to 'Yoast SEO Premium'. */
+			__( "To be able to create a redirect and fix this issue, you need %1$s. ", "wordpress-seo" ),
+			"Yoast SEO Premium"
+		);
+
+		const callToAction =  sprintf(
+			/* Translators: %1$s: yoast.com. */
+			__( "You can buy the plugin, including one year of support and updates, on %1$s.", "wordpress-seo" ),
+			"yoast.com"
+		);
+
+		const modalAriaLabel = sprintf(
+			/* translators: %s expands to 'Yoast SEO Premium'. */
+			__( "Get %s", "wordpress-seo" ),
+			"Yoast SEO Premium"
+		);
+
+		return (
+			<YoastModal
+				isOpen={ this.state.modalIsOpen }
+				onClose={ this.closeModal }
+				modalAriaLabel={ modalAriaLabel }
+				appElement={ document.querySelector( "#wpwrap" ) }
+				heading={ heading }
+			>
+				<StyledContainer>
+					<StyledIcon icon={ YoastSeoIcon } />
+					<h2>{ title }</h2>
+					<UpsellBox
+						infoParagraphs={ [ intro, callToAction ] }
+
+						upsellButtonText={
+							sprintf(
+								/* translators: %s expands to 'Yoast SEO Premium'. */
```

Comment:
You can use `{ heading }` instead

Reply:
Maybe rename `const heading` to `const upsellText` and use it for the heading, modalAriaLabel, and upsellButtonText. Currently, there are 3 identical translations (which translate.wordpress.org can handle, but leads to a lot of duplication in the codebase).